lib/graphql/fragment_cache/field_extension.rb

```diff
@@ -28,7 +28,8 @@ def build_extension(options)
       end
 
       def initialize(options:, **_rest)
-        @cache_options = options || {}
+        @cache_options = GraphQL::FragmentCache.default_options.merge(options || {})
```

Comment:
I might be missing something, but won't it work if we just do this in the `ObjectHelpers#cache_fragment` instead of checking for `default_options_merged`?

Reply:
That was my initial thought as well, but `context_key`, `cache_key`, and `if`/`unless`'s proc form are currently all unique to `GraphQL::FragmentCache::FieldExtension#resolve`: https://github.com/DmitryTsepelev/graphql-ruby-fragment_cache/blob/master/lib/graphql/fragment_cache/field_extension.rb#L33-L56.

So if we only merged options in `ObjectHelpers#cache_fragment`, then the following example wouldn't be possible:

```ruby
config.default_options = {
  cache_key: :object,
  if: -> { current_user.nil? }
}
```

A cleaner way may be to refactor `GraphQL::FragmentCache::FieldExtension#resolve` by removing most of its logic make it just delegate to `ObjectHelpers#cache_fragment`. And move all the `cache_key`/ `context_key` / `if`/`unless` processing to `ObjectHelpers#cache_fragment`. This way, no need to merge options twice, since everything is happening inside one method.

So I did this `default_options_merged` hack as a minimum effort that required no big refactor 😅, but let me know what you prefer or if you have any better idea!

Reply:
Ah, I thought no one would need global `if` but your example makes a total sense